example5.ts

```diff
@@ -0,0 +1,80 @@
+import * as readline from 'readline';
+
+type Position = {
```

Comment:
Consider using interfaces instead of type aliases for DTOs. Interfaces allow for better extensibility and declaration merging. I find your lack of interfaces disturbing.